src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3041,11 +3041,6 @@ private BoundStatement BindReturn(ReturnStatementSyntax syntax, BindingDiagnosti
                 hasErrors |= arg.HasErrors || ((object)arg.Type != null && arg.Type.IsErrorType());
             }
 
-            if (hasErrors)
-            {
-                return new BoundReturnStatement(syntax, refKind, BindToTypeForErrorRecovery(arg), @checked: CheckOverflowAtRuntime, hasErrors: true);
```

Comment:
>return new BoundReturnStatement(syntax, refKind, BindToTypeForErrorRecovery(arg), @checked: CheckOverflowAtRuntime, hasErrors: true); [](http://example.com/codeflow?start=16&length=133)

Consider suppressing any additional diagnostics by setting ```diagnostics``` value to "suppressed" instead of getting rid of the `if` completely.